repos/spack_repo/builtin/packages/variorum/package.py

```diff
@@ -53,13 +68,16 @@ class Variorum(CMakePackage):
     depends_on("fortran", type="build")  # generated
 
     depends_on("cmake@2.8:", type="build")
-    depends_on("hwloc")
     depends_on("jansson", type="link")
 
-    #########################
-    # Documentation related #
-    #########################
-    depends_on("py-sphinx", when="+docs", type="build")
+    # cuda@10.1.243 works, as does 12.4.1
+
+    depends_on("cuda", type=("build", "link"), when="gpu=nvidia")  # required for nvml
+    depends_on("esmi", type=("build", "link"), when="gpu=amd")      # required for amd
```

Comment:
@slabasan ESMI is needed for AMD CPUs. ROCM is needed for AMD GPUs. So the `when` here needs to be corrected.

Reply:
Will this fix do it?
```
depends_on("esmi", type=("build", "link"), when="cpu=amd")
depends_on("rocm", type=("build", "link"), when="gpu=amd")
```